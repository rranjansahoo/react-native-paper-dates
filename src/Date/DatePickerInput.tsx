import * as React from 'react'

import { TextInput } from 'react-native-paper'
import DatePickerModal from './DatePickerModal'
import { useLatest } from '../utils'
import type { DatePickerInputProps } from './DatePickerInput.shared'
import DatePickerInputWithoutModal from './DatePickerInputWithoutModal'
import { StyleProp, ViewStyle } from 'react-native'

function DatePickerInput(
  {
    withModal = true,
    calendarIcon = 'calendar',
    animationType = 'slide',
    ...rest
  }: DatePickerInputProps,
  ref: any
) {
  const [visible, setVisible] = React.useState<boolean>(false)
  const onDismiss = React.useCallback(() => {
    setVisible(false)
  }, [setVisible])
  const onChangeRef = useLatest(rest.onChange)
  const onInnerConfirm = React.useCallback(
    ({ date }: any) => {
      setVisible(false)
      onChangeRef.current(date)
    },
    [setVisible, onChangeRef]
  )

  return (
    <DatePickerInputWithoutModal
      ref={ref}
      {...rest}
      inputButton={
        withModal ? (
          <TextInput.Icon
            size={rest.iconSize ?? 24}
            icon={calendarIcon}
            color={rest.iconColor ?? undefined}
            disabled={rest.disabled}
            onPress={() => setVisible(true)}
            style={rest.iconStyle as StyleProp<ViewStyle>}
            testID={`${rest.testID || 'date-picker'}-icon-button`}
          />
        ) : null
      }
      // eslint-disable-next-line react/no-unstable-nested-components
      modal={({
        value,
        locale,
        inputMode,
        validRange,
        saveLabel,
        saveLabelDisabled,
        uppercase,
        startYear,
        endYear,
        inputEnabled,
        disableStatusBarPadding,
      }) =>
        withModal ? (
          <DatePickerModal
            date={value}
            mode="single"
            visible={visible}
            onDismiss={onDismiss}
            onConfirm={onInnerConfirm}
            locale={locale}
            dateMode={inputMode}
            validRange={validRange}
            saveLabel={saveLabel}
            saveLabelDisabled={saveLabelDisabled ?? false}
            uppercase={uppercase ?? true}
            startYear={startYear ?? 1800}
            endYear={endYear ?? 2200}
            inputEnabled={inputEnabled}
            disableStatusBarPadding={disableStatusBarPadding ?? false}
            animationType={animationType}
          />
        ) : null
      }
    />
  )
}

export default React.forwardRef(DatePickerInput)
