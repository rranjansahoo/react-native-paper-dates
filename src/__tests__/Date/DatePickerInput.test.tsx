import * as React from 'react'

import renderer from 'react-test-renderer'
import DatePickerInput from '../../Date/DatePickerInput'
import { SafeAreaProvider } from 'react-native-safe-area-context'

it('renders DatePickerInput', () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <DatePickerInput
          locale={'en'}
          value={new Date('12/26/2022')}
          onChange={() => null}
          inputMode="start"
          autoComplete={'birthdate-full'}
        />
      </SafeAreaProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
