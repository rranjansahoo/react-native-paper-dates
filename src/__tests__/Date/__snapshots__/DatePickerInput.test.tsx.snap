// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DatePickerInput 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
